test/e2e_node/services/server.go

```diff
@@ -282,34 +283,28 @@ func (s *server) kill() error {
 	}
 
 	// Attempt to shut down the process in a friendly manner before forcing it.
-	waitChan := make(chan error)
 	go func() {
 		_, err := cmd.Process.Wait()
-		waitChan <- err
-		close(waitChan)
 	}()
 
 	const timeout = 10 * time.Second
 	for _, signal := range []string{"-TERM", "-KILL"} {
 		klog.V(2).Infof("Killing process %d (%s) with %s", pid, name, signal)
-		cmd := exec.Command("kill", signal, strconv.Itoa(pid))
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
```

Comment:
Remove this defer:
a) it's not good to defer in a loop like this (the `defer`s get executed when the *function* exits, not when the current loop iteration exits)
b) this cancel function never needs to be called. When we pass the CommandContext function it's either finished, or timed out, in either case we don't need to cancel.

Reply:
Yeah it makes sense. Thanks for pointing out. 
We can do: 

`ctx, _ := context.WithTimeout(context.Background(), 5*time.Second)`